js/dsx.js

```diff
@@ -201,532 +250,262 @@ module.exports = class dsx extends Exchange {
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.privateGetOrder (this.extend (request, params));
+        return this.parseOrders (response);
     }
 
-    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+    async fetchOrder (id, symbol = undefined, params = {}) {
         await this.loadMarkets ();
-        const market = this.market (symbol);
         const request = {
-            'pair': market['id'],
+            'clientOrderId': id,
         };
+        const response = await this.privateGetHistoryOrder (this.extend (request, params));
+        return this.parseOrders (response);
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        if (symbol !== undefined) {
+            const market = this.market (symbol);
+            request['symbol'] = market['id'];
+        }
         if (limit !== undefined) {
             request['limit'] = limit;
         }
-        const response = await this.publicGetTradesPair (this.extend (request, params));
-        if (Array.isArray (response)) {
-            const numElements = response.length;
-            if (numElements === 0) {
-                return [];
-            }
+        if (since !== undefined) {
+            request['from'] = since;
         }
-        return this.parseTrades (response[market['id']], market, since, limit);
+        const response = await this.privateGetHistoryOrder (this.extend (request, params));
+        return this.parseOrders (response);
     }
 
-    parseOHLCV (ohlcv, market = undefined, timeframe = '1m', since = undefined, limit = undefined) {
-        //
-        //     {
-        //         "high" : 0.01955,
-        //         "open" : 0.01955,
-        //         "low" : 0.01955,
-        //         "close" : 0.01955,
-        //         "amount" : 2.5,
-        //         "timestamp" : 1565155740000
-        //     }
-        //
-        return [
-            this.safeInteger (ohlcv, 'timestamp'),
-            this.safeFloat (ohlcv, 'open'),
-            this.safeFloat (ohlcv, 'high'),
-            this.safeFloat (ohlcv, 'low'),
-            this.safeFloat (ohlcv, 'close'),
-            this.safeFloat (ohlcv, 'amount'),
-        ];
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.fetchOrders (symbol, since, limit, params);
+        const orders = this.filterBy (this.orders, 'status', 'closed');
+        return this.filterBySymbolSinceLimit (orders, symbol, since, limit);
     }
 
-    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
```

Comment:
@npomfret We could save this method by a replacing `pair` with `instrument` in it. What do you say?

Reply:
I'd suggest to use url param names as they're named in their API docs – for less confusion with params overrides.

Reply:
Does the new API have that feature?

Reply:
https://api.dsxglobal.com/#candles ?

Reply:
I think the old API is completely dead anyway, so merging and adding this method later is an option.

Reply:
Have added fetchOHLCV.  Looks like DSX is VERY similar, or possibly the same as hitbtc.  I would be nice to get some code reuse, but I don't think inheritance (`dsx extends hitbtc`) is the answer though, it inevitably leads to as many problems as it solves.  Perhaps if they both inherited from a common super class...

`class hitbtc extends AbstractHitbtcAndDsxFunctionality {...`

`class dsx extends AbstractHitbtcAndDsxFunctionality {...`

Or better (IMO), could we move the common functions out and have each implementation delegate by cherry picking which functions it wants to use?